include/mesh/mesh_base.h

```diff
@@ -852,6 +854,8 @@ class MeshBase : public ParallelObject
   virtual const_node_iterator local_nodes_end    () const = 0;
   virtual const_node_iterator pid_nodes_begin    (const processor_id_type proc_id) const = 0;
   virtual const_node_iterator pid_nodes_end      (const processor_id_type proc_id) const = 0;
+  virtual const_node_iterator bid_nodes_begin    (const boundary_id_type bndry_id) const = 0;
```

Comment:
It only iterates through nodes for which 

I agree that all these iterator methods should probably be documented.  I was thinking about reorganizing them so the const and non-const versions are together and we can put one 
```
/**
*
*/
```
section that covers all 4.